### PR TITLE
Feat vk abstractions

### PR DESCRIPTION
**Pull request by Kyn21kx**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
-# Unamed engine
+# Hush Engine
 
-TODO:
+Hush is a 3D game engine currently in development that aims to provide a more streamlined, simple and performat way to make high quality games
```
Comment (Kyn21kx) [inline README.md:3]: Oops, this was supposed to go on another branch to update the engine's info, but bled out here

Comment (Alan5142) [reply README.md:3]: Sure, no problem.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,18 @@
...
+
+**HUSH IS STILL IN EARLY STAGES OF DEVELOPMENT, A ROADMAP IS AVAILABLE [HERE](https://hushengine.com/)**
+
+## Features
+- 3D rendering (Vulkan)
+- C# scripting
+- Rust scripting **(WIP)**
+- C++ 17 support
```
Comment (Alan5142) [inline README.md:11]: We should update this to C++20

File: src/engine_core/rendering/src/Shared/RenderableNode.hpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+#include "IRenderable.hpp"
+namespace Hush {
+	/// @brief Common renderable node for scenes with multiple children to render
+	/// the Draw function *MUST* be called (recursed down) for every implementation
+	/// PENDING: Unite this into one renderableNode implementation IF AND ONLY IF this becomes a bottleneck
+	class RenderableNode : public IRenderable {
+
+	public:
+		void RefreshTransform(const glm::mat4& parentMatrix) {
+			this->m_worldTransform = parentMatrix * this->m_localTransform;
```
Comment (Alan5142) [inline src/engine_core/rendering/src/Shared/RenderableNode.hpp:15]: For this, we should look into integrating an ECS library, we could have a system that performs this. That way, we avoid coupling the rendering node (which should only contain data) with updating the transforms, also, the nodes become data-only containers, splitting the functionality in two, allowing more flexibility

File: src/engine_core/rendering/src/Shared/RenderableNode.hpp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+		const glm::mat4& GetLocalTransform() const noexcept {
+			return this->m_localTransform;
+		}
+
+		const glm::mat4& GetWorldTransform() const noexcept {
+			return this->m_worldTransform;
+		}
+
+	protected:
+		std::weak_ptr<RenderableNode> m_parent;
```
Comment (Alan5142) [inline src/engine_core/rendering/src/Shared/RenderableNode.hpp:44]: With the previous approach, tracking their parents should not be needed in this component, it only contains data (i.e. the local and world transform).

**Review by Alan5142 — APPROVED** [review]
LGTM

File: src/engine_core/rendering/src/Shared/RenderableNode.hpp (hunk 3 above)
Comment (Kyn21kx) [reply src/engine_core/rendering/src/Shared/RenderableNode.hpp:15]: I agree, this behaviour should not be a part of the rendering itself, eventually only the world transform should be passed as part of a render context and that's it, this is a temporary workaround to having updating transforms and children, but we should absolutely consider integrating ECS soon

**Pull request merged**